bevy_lint/src/driver.rs

```diff
@@ -0,0 +1,13 @@
+//! Contains code related to writing `bevy_lint_driver`.
+
+use crate::callback::BevyLintCallback;
+
+/// This is the main entrypoint into the driver, exported so that `bevy_cli` may call it.
+pub fn main() {
+    let args: Vec<String> = dbg!(std::env::args().skip(1).collect());
+
+    // Call the compiler with our custom callback.
+    rustc_driver::RunCompiler::new(&args, &mut BevyLintCallback)
+        .run()
+        .unwrap()
```

Comment:
Could `main` return a result here?

Reply:
Good idea! `run()` returns an [`ErrorGuaranteed`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_errors/struct.ErrorGuaranteed.html) on failure, which is equivalent to the unit `()`, so for now the function returns `Result<(), ()>`.